### PR TITLE
Implement minimum validations in UserMessage

### PR DESCRIPTION
**Pull request by rapala61**
#### Validations for the `/api/v1/events/user-create` route.

It validates only a minimum set of required properties that the messaging team is expecting for successful segmentation. All other properties sent, aside from this subset of required props, are considered optional or nullable. The Required properties are:
- id (northstar_id)
- updated_at
- created_at
- email OR mobile

## How to test
- Run the tests and get no errors:
    - `cd` into your local Blink repo.
    - `yarn install`
    - `docker-compose up -d`
    - `yarn test:full`
    - `docker-compose down`

## Relevant Issues
Fixes `last_messaged_at` bug [Pivotal #152901264](https://www.pivotaltracker.com/story/show/152901264)
Fixes `sms_paused` bug [Pivotal #153457736](https://www.pivotaltracker.com/story/show/153457736)

## Other tasks
- [x] Deploy to staging.
- [x] Going through an user creation/update and inspecting the events generated for the **staging** user in the **staging** C.io instance.
- [x] Update [UserMessage Schema](https://github.com/DoSomething/blink/wiki/Message-Schemas#usermessage) Wiki docs.

### CODE REVIEW

**Comment by rapala61** [comment]
My `last_messaged_at` user property was updated successfully in C.io. The `sms_paused` however, seems like it's not being sent from Northstar.

**Comment by rapala61** [comment]
Did you manage to see the `sms_paused` property in C.io? @aaronschachter

**Comment by aaronschachter** [comment]
@rapala61 -- Confirmed that I do *not* see any updates to Customer.io Staging for `sms_paused`

File: src/validations/userCreateAndUpdate.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  created_at: Joi.string().empty(whenNullOrEmpty).required().isoDate(),
+
+  // Remove field when provided as empty string or null.
+  email: Joi.string().empty(whenNullOrEmpty).default(undefined),
+  mobile: Joi.string().empty(whenNullOrEmpty).default(undefined),
+
+  // Although this is NOT required to be sent, we make it implicitly required to exist in
+  // the payload we send to C.io by declaring a default.
+  // Allow anything as a role, but default to user.
+  role: Joi.string().empty(whenNullOrEmpty).default('user'),
```
Comment (DFurnes) [inline src/validations/userCreateAndUpdate.js]: FWIW, this should always default to `user` in the payload sent by Northstar.

File: src/messages/CustomerIoUpdateCustomerMessage.js [hunk 2]
```diff
@@ -99,8 +42,14 @@ class CustomerIoUpdateCustomerMessage extends Message {
       ).unix();
```
Comment (DFurnes) [inline src/messages/CustomerIoUpdateCustomerMessage.js]: It'd be nice to update these to accept both timestamps & ISO-8601 dates (so we can start sending dates as timestamps across the board in Northstar). Would something like this work?

```js
if (! Number.isInteger(customerData.created_at)) {
  customerData.created_at = moment(customerData.created_at, moment.ISO_8601).unix();
}
```

**Comment by rapala61** [comment]
Thanks @aaronschachter. @DFurnes confirmed that the `sms_paused` was actually not ready on their end, but created a PR right away for it https://github.com/DoSomething/northstar/pull/707. We should be getting the property in C.io once it makes its way to Production.

File: src/messages/CustomerIoUpdateCustomerMessage.js (hunk 2 above)
Comment (rapala61) [reply src/messages/CustomerIoUpdateCustomerMessage.js]: I think that should definitely work! 👍 Thanks.

**Comment by rapala61** [comment]
@DFurnes I'm testing in staging and I can see an update sent to C.io when the user goes into paused mode `sms_paused: true`, but when the user switches to `sms_paused:false`, the updated value is not sent to Blink from Northstar. 🤔

**Comment by DFurnes** [comment]
Oh, hmm. It looks like we're sending `null` when that value is false. That's not right. 😖

**Comment by DFurnes** [comment]
@rapala61 Caught the issue & pushed up a quick fix in DoSomething/northstar#708.

Also just want to confirm - when we send `null` for a field to Customer.io user profiles, it's being unset on the corresponding C.io profile right? For example, thinking of a scenario where a user removes their phone number from their account & we'd want to make sure that was nulled out on C.io as well.

**Comment by rapala61** [comment]
Thank you @DFurnes ! [That](https://github.com/DoSomething/northstar/pull/708) worked 👍 . 

>when we send null for a field to Customer.io user profiles, it's being unset on the corresponding C.io profile right?

Not with these changes, since we are removing a lot of the more strict validations and the default is to not allow `null` values. Thanks for the 💭 ! We have to allow `null` values for user profile creation and updates. I'll include that in a commit before merging this PR!

**Comment by rapala61** [comment]
@DFurnes I can confirm that setting a field to `null` in Northstar, replicates the change to C.io.

**Pull request merged**